homeassistant/components/microbot_push/config_flow.py

```diff
@@ -0,0 +1,206 @@
+        self._discovered_adv = parsed
+        data = parsed.data
+        self.context["title_placeholders"] = {
+            "name": data["local_name"],
+            "address": discovery_info.address,
+        }
+        return await self.async_step_init()
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initialized by the user."""
+        # This is for backwards compatibility.
+        return await self.async_step_init(user_input)
+
+    async def async_step_init(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Check if paired."""
+        errors: dict[str, str] = {}
+
+        if discovery := self._discovered_adv:
+            self._discovered_advs[discovery.address] = discovery
+        else:
+            current_addresses = self._async_current_ids()
+            for discovery_info in async_discovered_service_info(self.hass):
+                address = discovery_info.address
+                if (
+                    format_unique_id(address) in current_addresses
+                    or address in self._discovered_advs
+                ):
+                    continue
+                parsed = parse_advertisement_data(
+                    discovery_info.device, discovery_info.advertisement
+                )
+                if parsed:
+                    self._discovered_advs[address] = parsed
+
+        if not self._discovered_advs:
+            return self.async_abort(reason="no_unconfigured_devices")
+
+        data_schema = vol.Schema(
+            {
+                vol.Required(CONF_BDADDR): vol.In(
+                    {
+                        address: f"{parsed.data['local_name']} ({address})"
+                        for address, parsed in self._discovered_advs.items()
+                    }
+                ),
+                vol.Required(CONF_NAME): str,
+            }
+        )
+
+        if user_input is not None:
+            self._name = user_input[CONF_NAME]
+            self._bdaddr = user_input[CONF_BDADDR]
+            await self.async_set_unique_id(
+                format_unique_id(self._bdaddr), raise_on_progress=False
+            )
+            self._abort_if_unique_id_configured()
+            self._ble_device = bluetooth.async_ble_device_from_address(
+                self.hass, self._bdaddr.upper()
+            )
+            if not self._ble_device:
+                raise ConfigEntryNotReady(
+                    f"Could not find MicroBot with address {self._bdaddr}"
+                )
+            conf = (
+                self.hass.config.path()
+                + "/.storage/microbot-"
+                + re.sub("[^a-f0-9]", "", self._bdaddr.lower())
+                + ".conf"
+            )
+            self._client = MicroBotApiClient(
+                device=self._ble_device,
+                config=conf,
+                retry_count=DEFAULT_RETRY_COUNT,
+            )
```

Comment:
This is doing I/O which will block the event loop

https://github.com/spycle/pyMicroBot/blob/32700e5eabf38814bcb26fc46cb0388a080ae450/microbot/__init__.py#L161

Please store the tokens in the config entry data instead